tests/db_functions/text/test_right.py

```diff
@@ -26,6 +26,14 @@ def test_invalid_length(self):
         with self.assertRaisesMessage(ValueError, "'length' must be greater than 0"):
             Author.objects.annotate(raises=Right("name", 0))
 
+    def test_zero_length(self):
+        authors = Author.objects.annotate(
+            name_part=Right("name", Value(0, output_field=IntegerField()))
+        )
+        self.assertQuerySetEqual(
+            authors.order_by("name"), ["", ""], lambda a: a.name_part
+        )
```

Comment:
I don't believe that anyone would pass `Value(0)` as a length. I'd use a "real-life" example, e.g.
```python
    def test_zero_length(self):
        Author.objects.create(name="Tom", alias="tom")
        authors = Author.objects.annotate(
            name_part=Right("name", Length("name") - Length("alias")),
        )
        self.assertQuerySetEqual(
            authors.order_by("name"),
            [
                "mith",
                "" if connection.features.interprets_empty_strings_as_nulls else None,
                "",
            ],
            lambda a: a.name_part,
        )
```

Reply:
_"Blessed are those who have not seen, and have believed."_

You are right, I suggested myself with the existing tests. Your example looked good so I copied it.